controllers/cluster_metrics.go

```diff
@@ -0,0 +1,64 @@
+package controllers
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	metricNamespace = "upgrade_schedule_operator"
+	metricSubsystem = "cluster"
+)
+
+// Counters for total applied and failed scheduled upgrades
+var (
+	counterLabels = []string{"cluster_id", "cluster_namespace", "origin_version", "target_version"}
+
+	UpgradesTotal = prometheus.NewCounterVec(
```

Comment:
They should be equal. We increment the `total` each time we try to apply the upgrade. We increment the `failure` in case such an application fails. In case of success we increment the `success`. The only thing to keep in mind is that we will try to apply the upgrade again in case it failed in the last reconciliation loop so we could get a high percentage of failures if we don't account for retries in our statistic.

Reply:
ah yes! I'd misread the code and thought this was incremented prior to running the upgrade.